managed/models/encryption_helpers.go

```diff
@@ -0,0 +1,308 @@
+		if err != nil {
+			logrus.Warning(err)
+		}
+		agent.PostgreSQLOptions.SSLKey, err = handler(agent.PostgreSQLOptions.SSLKey)
+		if err != nil {
+			logrus.Warning(err)
+		}
+	}
+
+	if agent.MongoDBOptions != nil {
+		agent.MongoDBOptions.TLSCertificateKey, err = handler(agent.MongoDBOptions.TLSCertificateKey)
+		if err != nil {
+			logrus.Warning(err)
+		}
+		agent.MongoDBOptions.TLSCertificateKeyFilePassword, err = handler(agent.MongoDBOptions.TLSCertificateKeyFilePassword)
+		if err != nil {
+			logrus.Warning(err)
+		}
+	}
+
+	if agent.AzureOptions != nil {
+		agent.AzureOptions.ClientID, err = handler(agent.AzureOptions.ClientID)
+		if err != nil {
+			logrus.Warning(err)
+		}
+		agent.AzureOptions.ClientSecret, err = handler(agent.AzureOptions.ClientSecret)
+		if err != nil {
+			logrus.Warning(err)
+		}
+		agent.AzureOptions.SubscriptionID, err = handler(agent.AzureOptions.SubscriptionID)
+		if err != nil {
+			logrus.Warning(err)
+		}
+		agent.AzureOptions.TenantID, err = handler(agent.AzureOptions.TenantID)
+		if err != nil {
+			logrus.Warning(err)
+		}
+	}
+
+	return agent
+}
+
+// EncryptMySQLOptionsHandler returns encrypted MySQL Options.
+func EncryptMySQLOptionsHandler(e *encryption.Encryption, val any) (any, error) {
+	return mySQLOptionsHandler(val, e.Encrypt)
+}
+
+// DecryptMySQLOptionsHandler returns decrypted MySQL Options.
+func DecryptMySQLOptionsHandler(e *encryption.Encryption, val any) (any, error) {
+	return mySQLOptionsHandler(val, e.Decrypt)
+}
+
+func mySQLOptionsHandler(val any, handler func(string) (string, error)) (any, error) {
+	o := MySQLOptions{}
+	value := val.(*sql.NullString) //nolint:forcetypeassert
+	if !value.Valid {
+		return sql.NullString{}, nil
+	}
+
+	err := json.Unmarshal([]byte(value.String), &o)
+	if err != nil {
+		return nil, err
+	}
+
+	o.TLSCa, err = handler(o.TLSCa)
+	if err != nil {
+		return nil, err
+	}
+	o.TLSCert, err = handler(o.TLSCert)
+	if err != nil {
+		return nil, err
+	}
+	o.TLSKey, err = handler(o.TLSKey)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := json.Marshal(o)
+	if err != nil {
+		return nil, err
+	}
+
+	return res, nil
+}
+
+// EncryptPostgreSQLOptionsHandler returns encrypted PostgreSQL Options.
+func EncryptPostgreSQLOptionsHandler(e *encryption.Encryption, val any) (any, error) {
+	return postgreSQLOptionsHandler(val, e.Encrypt)
+}
+
+// DecryptPostgreSQLOptionsHandler returns decrypted PostgreSQL Options.
+func DecryptPostgreSQLOptionsHandler(e *encryption.Encryption, val any) (any, error) {
+	return postgreSQLOptionsHandler(val, e.Decrypt)
+}
+
+func postgreSQLOptionsHandler(val any, handler func(string) (string, error)) (any, error) {
+	o := PostgreSQLOptions{}
+	value := val.(*sql.NullString) //nolint:forcetypeassert
+	if !value.Valid {
+		return sql.NullString{}, nil
+	}
+
+	err := json.Unmarshal([]byte(value.String), &o)
+	if err != nil {
+		return nil, err
+	}
+
+	o.SSLCa, err = handler(o.SSLCa)
```

Comment:
@JiriCtvrtka @BupycHuk 
Do we need to encrypt the `Certificate Authority` (aka CA) fields, given that they are meant to be public? Or is it not always the case?

Reply:
It was leftover in handler, but it is removed.